common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -51,3 +54,105 @@ def perform_destroy(self, instance):
             )
         if save_model:
             instance.save()
+
+
+class EntitlementEnrollmentViewSet(viewsets.GenericViewSet):
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = CourseEntitlement.objects.all()
+    serializer_class = CourseEntitlementSerializer
+
+    def _enroll_entitlement(self, entitlement, course_session_key, user):
+        enrollment = CourseEnrollment.enroll(
+            user=user,
+            course_key=course_session_key,
+            mode=entitlement.mode,
+        )
```

Comment:
Does this save to the DB immediately? If so, it should probably be run in a transaction along with https://github.com/edx/edx-platform/pull/16617/files#diff-9b73749954d13e21a49b7ea1e0b110a9R72